Mapbox Mobile Events 1.0.0-beta1

alfwatt opened the pull request:
→ Add MMEVersions.h to store the version info
→ Convert all the #pragma marks
→ I very intentionally MMECommonEventData
→ Remove MMETimerManager and it's tests
→ Remove MMENamespaceDependencies
→ Only validate bundle version if it's declared as semver

Commit:
Mapbox Mobile Events 1.0-beta1

-> Add MMEVersions.h to store the version info
-> Convert all the #pragma marks
-> I very intentionally MMECommonEventData
-> Remote the MMETimerManager and it's tests, along with MMENamespaceDependencies
-> Only validate the bundle version if it's declared as a semver string

rclee commented on Tests/MMEEventsManagerTests.m:
```diff
@@ -14,7 +14,7 @@
 
 #import "MMEAPIClientFake.h"
-#import "MMETimerManagerFake.h"
+//#import "MMETimerManagerFake.h"
```
Good to kill this and it's relatives?  🔪

rclee commented on Tests/MMEEventsManagerTests.mm:
```diff
@@ -281,28 +277,28 @@ @interface MMEEvent (Tests)
-                        eventsManager.apiClient should_not have_received(@selector(postEvents:completionHandler:));
-                    });
-                });
+//                context(@"when there are events in the queue", ^{
```
We should reimplement these in XCTest probably

alfwatt commented on MMEVersions.h:
```diff
@@ -0,0 +1,7 @@
+#define MMEVersions_h
+
+#define MME_BUNDLE_VERSION 100.0
+#define MME_SHORT_VERSION_STRING 1.0.0-dev
```
1.0.0-beta

Commit:
Add a target and scheme for building with deprecation warnings enabled
Fix deprecation tag in MMEEventLogger.h

Commit:
Better target name

Commit:
MMEEventLogger → MMELogger
Add tests for MMELogger
Add XCTests to the Deprecation scheme (Cedar tests are Deprecated)
Set log level depending on DEBUG status
